Formula/flarectl.rb

```diff
@@ -0,0 +1,28 @@
+class Flarectl < Formula
+  desc "CLI application for interacting with a Cloudflare account"
+  homepage "https://github.com/cloudflare/cloudflare-go/tree/master/cmd/flarectl"
+  url "https://github.com/cloudflare/cloudflare-go/archive/v0.11.6.tar.gz"
+  sha256 "f3eafe5f5b891ed36c0dbfb76988cf5b5844700eea9444cbab9b85e287ba7a91"
+  head "https://github.com/cloudflare/cloudflare-go.git"
+
+  depends_on "go" => :build
+  depends_on "gox" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/cloudflare/cloudflare-go"
+    dir.install buildpath.children
+
+    cd dir do
+      arch = Hardware::CPU.is_64_bit? ? "amd64" : "386"
+      system "gox", "-arch", arch, "-os", "darwin", "-ldflags", "\"-X main.Rev=#{version}\"", "./..."
+      bin.install "flarectl_darwin_#{arch}" => "flarectl"
+    end
+
+    prefix.install_metafiles dir
+  end
+
+  test do
+    system "#{bin}/flarectl", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I did a new test, it will require connectivity, but I think takes into account what you were looking for.